pandas/io/formats/style.py

```diff
@@ -157,53 +157,40 @@ def __init__(
         na_rep: Optional[str] = None,
         uuid_len: int = 5,
     ):
-        self.ctx: DefaultDict[Tuple[int, int], List[str]] = defaultdict(list)
-        self._todo: List[Tuple[Callable, Tuple, Dict]] = []
-
+        # validate ordered args
         if not isinstance(data, (pd.Series, pd.DataFrame)):
             raise TypeError("``data`` must be a Series or DataFrame")
         if data.ndim == 1:
             data = data.to_frame()
         if not data.index.is_unique or not data.columns.is_unique:
             raise ValueError("style is not supported for non-unique indices.")
-
-        self.data = data
-        self.index = data.index
-        self.columns = data.columns
-
+        assert isinstance(data, DataFrame)
```

Comment:
this is superfluous, if you can remove in a followon (see L161)

Reply:
this was added otherwsise mypy complains about mismatched type. maybe there is a better way..?